File: youlexuanparent/youlexuan-shop-web/src/main/webapp/js/service/goodsService.js
//goods服务层
app.service('goodsService', function($http){
	// 保存、修改
	this.save = function(entity) {
		var methodName = 'add'; 	// 方法名称
		if (entity.goods.id != null) { 	// 如果有ID
			methodName = 'update'; 	// 则执行修改方法
		}
		return $http.post('../goods/' + methodName + '.do', entity);
	}

	// 查询单个实体
	this.findOne = function(id) {
		return $http.get('../goods/findOne.do?id=' + id);
	}

	// 批量删除
	this.dele = function(ids) {
		// 获取选中的复选框
		return $http.get('../goods/delete.do?ids=' + ids);
	}

	// 查询
	this.search = function(page, size, searchEntity) {
		// post提交，page、size属性和之前相同，将searchEntity提交至后台@RequestBody对应的属性
		return $http.post('../goods/search.do?page=' + page + '&size=' + size,
				searchEntity);
	}
	//增加基本属性
	this.add = function(entity) {
		return $http.post('../goods/add.do', entity);
	}
	//更改状态
	this.updateStatus=function(ids,status){
		return $http.get('../goods/updateStatus.do?ids='+ids+"&status="+status);
	}
});

File: youlexuanparent/youlexuan-sellergoods-interface/src/main/java/com/sunxupeng/sellergoods/service/GoodsService.java
package com.sunxupeng.sellergoods.service;

import com.sunxupeng.entity.PageResult;
import com.sunxupeng.group.Goods;
import com.sunxupeng.pojo.TbGoods;
import com.sunxupeng.pojo.TbItem;

import java.util.List;

/**
 * goods服务层接口
 * @author sunxupeng
 *
 */
public interface GoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(Goods goods);


	/**
	 * 修改
	 */
	public void update(Goods goods);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Goods findOne(Long id);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbGoods goods, int pageNum, int pageSize);

	 //增加，传递组合参数

	//批量修改状态
	 public void updateStatus(Long []ids, String status);

	 public List<TbItem> findItemListByGoodsId(Long[]ids,String status);


}
